airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/data/iceberg_parquet/AirbyteTypeToIcebergSchema.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data.iceberg_parquet
+
+import io.airbyte.cdk.load.data.AirbyteType
+import io.airbyte.cdk.load.data.ArrayType
+import io.airbyte.cdk.load.data.ArrayTypeWithoutSchema
+import io.airbyte.cdk.load.data.BooleanType
+import io.airbyte.cdk.load.data.DateType
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.NumberType
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.data.ObjectTypeWithEmptySchema
+import io.airbyte.cdk.load.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.TimeTypeWithTimezone
+import io.airbyte.cdk.load.data.TimeTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithoutTimezone
+import io.airbyte.cdk.load.data.UnionType
+import io.airbyte.cdk.load.data.UnknownType
+import java.util.UUID
+import org.apache.iceberg.Schema
+import org.apache.iceberg.types.Type
+import org.apache.iceberg.types.Types
+import org.apache.iceberg.types.Types.NestedField
+
+class AirbyteTypeToIcebergSchema {
+
+    fun convert(airbyteSchema: AirbyteType): Type {
+        return when (airbyteSchema) {
+            is ObjectType -> {
+                Types.StructType.of(
+                    *airbyteSchema.properties.entries
+                        .map { (name, field) ->
+                            if (field.nullable) {
+                                NestedField.optional(
+                                    UUID.randomUUID().hashCode(),
```

Comment:
is this the field ID? don't we need this to be stable across syncs, i.e. not randomized?

Reply:
For the first sync this would be used, for the subsequent syncs the schema loaded from the iceberg table would be used. We just have to make sure that the schema stays the same.